backend/app/instance_provider.go

```diff
@@ -67,6 +67,10 @@ func (ip *instanceProvider) NeedsUpdate(_ context.Context, pluginContext backend
 	curAppSettings := pluginContext.AppInstanceSettings
 	appUpdated := !curAppSettings.Updated.Equal(cachedAppSettings.Updated)
 
+	if appUpdated || configUpdated {
```

Comment:
Since you care about Prometheus, this should be in the datasource instance provider